File: app/src/main/java/com/globalwarming/earthsaver/ForgotPasswordActivity.java
package com.globalwarming.earthsaver;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;

import android.app.ProgressDialog;
import android.os.Bundle;
import android.widget.Button;
import android.widget.Toast;

import com.google.android.material.snackbar.Snackbar;
import com.google.android.material.textfield.TextInputLayout;
import com.google.firebase.auth.FirebaseAuth;

public class ForgotPasswordActivity extends AppCompatActivity {

    private Toolbar toolbar;
    private TextInputLayout etEmail;
    private Button button;
    private ProgressDialog progressDialog;

    private FirebaseAuth mAuth;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_forgot_password);

        toolbar = findViewById(R.id.toolbar);
        etEmail = findViewById(R.id.etEmail);
        button = findViewById(R.id.buttonResetPassword);
        setSupportActionBar(toolbar);
        toolbar.setNavigationOnClickListener(v -> {
            onBackPressed();
        });

        progressDialog = new ProgressDialog(this);
        progressDialog.setMessage("Resetting password");
        progressDialog.setCanceledOnTouchOutside(false);
        progressDialog.setCancelable(false);

        mAuth = FirebaseAuth.getInstance();

        button.setOnClickListener(v -> {
            String email = etEmail.getEditText().getText().toString().trim();
            if (!email.isEmpty()) {
                Util.hideKeyboard(v);
                progressDialog.show();

                mAuth.sendPasswordResetEmail(email).addOnCompleteListener(task -> {
                    progressDialog.dismiss();
                    if (task.isSuccessful()) {
                        Toast.makeText(ForgotPasswordActivity.this, "Reset link sent to registered email", Toast.LENGTH_LONG).show();
                        finish();
                    } else {
                        Snackbar.make(v, task.getException().getMessage(), Snackbar.LENGTH_SHORT).show();
                    }
                });
            }
        });

    }

    @Override
    protected void onDestroy() {
        if (progressDialog.isShowing()) {
            progressDialog.dismiss();
        }
        super.onDestroy();
    }

}

File: app/src/main/java/com/globalwarming/earthsaver/ForgotPasswordActivity.kt
package com.globalwarming.earthsaver

import android.os.Bundle
import android.widget.*
import androidx.appcompat.app.AppCompatActivity
import androidx.databinding.DataBindingUtil
import com.globalwarming.earthsaver.databinding.ActivityForgotPasswordBinding
import com.globalwarming.earthsaver.util.Loader
import com.google.android.gms.tasks.Task
import com.google.android.material.snackbar.Snackbar
import com.google.firebase.auth.FirebaseAuth

class ForgotPasswordActivity : AppCompatActivity() {

    private lateinit var binding: ActivityForgotPasswordBinding
    private val loader by lazy { Loader(this) }
    private lateinit var mAuth: FirebaseAuth

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_forgot_password)
        setSupportActionBar(binding.toolbar)

        mAuth = FirebaseAuth.getInstance()

        binding.toolbar.setNavigationOnClickListener { _ ->
            onBackPressed()
        }

        binding.buttonResetPassword.setOnClickListener { v ->
            val email = binding.etEmail.editText?.text?.toString()?.trim()
            if (!email.isNullOrEmpty()) {
                Util.hideKeyboard(v)
                loader.show()
                mAuth.sendPasswordResetEmail(email).addOnCompleteListener { task: Task<Void?>? ->
                    loader.dismiss()
                    if (task?.isSuccessful == true) {
                        Toast.makeText(
                            this@ForgotPasswordActivity,
                            "Reset link sent to registered email",
                            Toast.LENGTH_LONG
                        ).show()
                        finish()
                    } else {
                        Snackbar.make(v, task?.exception?.message.toString(), Snackbar.LENGTH_SHORT).show()
                    }
                }
            }
        }
    }

    override fun onDestroy() {
        if (loader.isShowing) {
            loader.dismiss()
        }
        super.onDestroy()
    }
}